crates/ruff_linter/src/rules/ruff/rules/unnecessary_dict_comprehension_for_iterable.rs

```diff
@@ -0,0 +1,192 @@
+        if self.is_value_none_literal {
+            format!(
+                "Unnecessary dict comprehension for iterable (rewrite using `dict.fromkeys(iterable)`)"
+            )
+        } else {
+            format!(
+                "Unnecessary dict comprehension for iterable (rewrite using `dict.fromkeys(iterable, value)`)"
+            )
+        }
+    }
+
+    #[allow(clippy::match_bool)]
+    fn fix_title(&self) -> String {
+        if self.is_value_none_literal {
+            format!("Rewrite using `dict.fromkeys(iterable, value)`)",)
+        } else {
+            format!("Rewrite using `dict.fromkeys(iterable)`)")
+        }
+    }
+}
+
+/// RUF025
+pub(crate) fn unnecessary_dict_comprehension_for_iterable(
+    checker: &mut Checker,
+    expr: &Expr,
+    key: &Expr,
+    value: &Expr,
+    generators: &[Comprehension],
+) {
+    let [generator] = generators else {
+        return;
+    };
+
+    // Don't suggest `dict.fromkeys` for async generator expressions, because `dict.fromkeys` is not async.
+    // Don't suggest `dict.fromkeys` for nested generator expressions, because `dict.fromkeys` might be error-prone option at least for fixing.
+    // Don't suggest `dict.fromkeys` for generator expressions with `if` clauses, because `dict.fromkeys` might not be valid option.
+    if !generator.ifs.is_empty() && generator.is_async && generators.len() > 1 {
+        return;
+    }
+    let Expr::Name(_) = &generator.target else {
+        return;
+    };
+
+    // Don't suggest `dict.fromkeys` if key and value are binded to the same name.
+    if let (Expr::Name(key_name), Expr::Name(value_name)) = (key, value) {
+        if key_name.id == value_name.id {
+            return;
+        }
+    }
+
+    if !has_valid_expression_type(value) {
+        return;
+    }
+
+    let mut diagnostic = Diagnostic::new(
+        UnnecessaryDictComprehensionForIterable {
+            is_value_none_literal: value.is_none_literal_expr(),
+        },
+        expr.range(),
+    );
+    diagnostic.try_set_fix(|| {
+        fix_unnecessary_dict_comprehension(expr, checker.locator(), checker.stylist())
+            .map(Fix::safe_edit)
+    });
+    checker.diagnostics.push(diagnostic);
+}
+
+// only accept `None`, `Ellipsis`, `True`, `False`, `Number`, `String`, `Bytes`, `Name` as value
+fn has_valid_expression_type(node: &ast::Expr) -> bool {
+    matches!(
+        node,
+        ast::Expr::StringLiteral(_)
+            | ast::Expr::BytesLiteral(_)
+            | ast::Expr::NumberLiteral(_)
+            | ast::Expr::BooleanLiteral(_)
+            | ast::Expr::NoneLiteral(_)
+            | ast::Expr::EllipsisLiteral(_)
+            | ast::Expr::Name(_)
+    )
+}
+
+/// Generate a [`Fix`] to replace `dict` comprehension with `dict.fromkeys`.
+/// (RUF025) Convert `{n: None for n in [1,2,3]}` to `dict.fromkeys([1,2,3])` or
+/// `{n: 1 for n in [1,2,3]}` to `dict.fromkeys([1,2,3], 1)`.
+fn fix_unnecessary_dict_comprehension(
```

Comment:
Is there a specific reason you choose to use `libcst` nodes instead of the [AST nodes](https://github.com/astral-sh/ruff/blob/993e605c6c57786acd836599a37f9e7c442b237b/crates/ruff_python_ast/src/nodes.rs) to construct the fixed node? I might be wrong but I think it might be easier to construct the node using the AST nodes with the `Generator` to generate the equivalent source code. I'm fine with either but just trying to understand this.

Reply:
No, specific reason. I was not quite sure what would be the correct way to construct the fixed node. Seems that using AST node with `Generator` would be more reasonable.